### DISCUSSION TITLE
Markdown support

### DISCUSSION THREAD

**Discussion by potomak** [zed-industries/zed#6622] [Feature Requests]
What is the plan for supporting Markdown?

There is already a syntax highlighter, but I don't think there is a formatter or other small utilities that would make easier to author Markdown in Zed, for instance automatically creating new list items on a new line after a list item, etc.

**Comment by jansol** [comment]
That particular example is tangentially related to #707 since the same mechanic is very nice for writing multi-line comments.

**Comment by JosephTLyons** [comment]
I think our first two steps for improving Markdown support will be:

1. Integrate an existing language server (either the VS Code one, or Marksman).  This will add a bunch of nice features, like completions, find all references, etc.
2. Add spell check

We have other ideas after that, but I think that those are the primary things we need to do.

At the moment, spell check is on our current dev cycle.  Not sure when we will get the language server integration - I'll ask the team about it.

**Reply by neurtaths-boat** [reply]
Why spell check? It seems so specific. I don't mean this passive aggressively I'm just genuinely curious why spell check itself has importance!

**Reply by JosephTLyons** [reply]
> Why spell check? It seems so specific. I don't mean this passive aggressively I'm just genuinely curious why spell check itself has importance!

I imagine it wouldn't be hardcoded to Markdown, there'd be some setting to determine what files it would be enabled for.  But I think most people would enable it for Markdown, which is why I included it here.

**Reply by jer-k** [reply]
Did anything ever come of the spellcheck work? Looks like on the Adoption roadmap it is further down the list.

**Reply by jansol** [reply]
zed-industries/extensions#691 has a WIP solution for spelling and writing style checking via [Vale](https://vale.sh/)

**Comment by gabrielalexandrelopes** [comment]
What about previewing Markdown in a panel with realtime update? Having preview capabilities not only for Markdown, but also HTML, SVG, plantUML, PDF, latex, etc, would really be helpful. Maybe future plugins?

**Reply by jansol** [reply]
The problem with HTML and SVG is that you basically need to implement a full web browser for those. PDF gets close to requiring that too, if you want to have *good* support for its features. And a full web browser is *humongous*.

**Reply by gabrielalexandrelopes** [reply]
yes, that makes sense. That's why Atom was so practical.

**Reply by gabrielalexandrelopes** [reply]
Well, for me the dream IDE would have:
- speed and simple UI (✔️ )
- interaction with language servers (✔️ )
- interaction with Jupyter kernels to run code snippets directly (like hydrogen in atom)
- git interaction (like atom or git cola)
- file preview with real time updates
- application debugging

that's why I'm still using atom/pulsar, but very excited about zed.

**Reply by Tom-Boscher** [reply]
Hydrogen was a really game changer for me!

**Reply by jansol** [reply]
Given the assistant panel that has been teased recently I imagine some very rudimentary Jupyter integration could actually be done with extensions eventually.

**Reply by 4rc0s** [reply]
Markdown preview would be very nice. I like it for project info and notes when learning new concepts. Thanks for considering.

**Reply by jansol** [reply]
Good news: [(limited) markdown rendering appears to be in the works](https://zed.dev/blog/zed-weekly-20). While this seems to be only for documentation tooltips so far, once the functionality is there I imagine it'd be fairly straightforward to reuse it for rendering full document previews.

**Reply by jansol** [reply]
(just noting here for posterity that live markdown preview has been out for a good while, search `markdown: open preview` in the command palette to open it)
